### PR TITLE
Issue 229 - add feature support for Project Lombok.

### PR DESCRIPTION
**Pull request by wetted**
Note that lombok must precede micronaut-inject-java for annotation processing.

### CODE REVIEW

**Comment by wetted** [comment]
This PR replaces https://github.com/micronaut-projects/micronaut-starter/pull/557

File: starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw [hunk 1]
```diff
@@ -819,6 +822,14 @@ List<Property> properties
           <!-- Uncomment to enable incremental compilation -->
           <!-- <useIncrementalCompilation>false</useIncrementalCompilation> -->
           <annotationProcessorPaths combine.children="append">
+@if (features.contains("lombok")) {
+            <path>
+              <!-- must precede micronaut-inject-java -->
```
Comment (alvarosanchez) [inline starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw:827]: This one is tricky. By default, now we inherit the annotation processor definitions from the parent POM, and the `combine.children="append"` attribute in line 824 will make the effective POM to contain [the parent elements](https://github.com/micronaut-projects/micronaut-core/blob/2.2.x/parent/build.gradle#L121-L130) first, then Lombok.

If Lombok needs to go first, then your feature needs to generate something like:

```xml
<annotationProcessorPaths combine.self="override">
  <!-- lombok -->
  <!-- micronaut-inject-java -->
  <!-- micronaut-validation -->
</annotationProcessorPaths>
```

File: starter-core/src/test/groovy/io/micronaut/starter/feature/other/ProjectLombokSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        def context = buildGeneratorContext(
+                ['lombok'],
+                new Options(Language.JAVA, TestFramework.SPOCK, BuildTool.MAVEN),
+                ApplicationType.DEFAULT)
+        String template = pom.template(ApplicationType.DEFAULT, buildProject(),
+                context.getFeatures(), context.getBuildProperties().getProperties()).render().toString()
+
+        then:
+        template.contains("""
+    <lombok.version>1.18.16</lombok.version>
```
Comment (alvarosanchez) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/other/ProjectLombokSpec.groovy:73]: Where is this coming from? If it's in the BOM, then the property will be part of the BOM's POM., and inherited to children. It should not be generated in the end-user POMs.

**Review by alvarosanchez — CHANGES_REQUESTED** [review]

File: starter-core/src/test/groovy/io/micronaut/starter/feature/other/ProjectLombokSpec.groovy (hunk 2 above)
Comment (wetted) [reply starter-core/src/test/groovy/io/micronaut/starter/feature/other/ProjectLombokSpec.groovy:73]: OK. It looks like we are adding similar properties for others like dekorate and openapi, so maybe we should look through other features to clean things up (unless I am missing some reason they need to be there).

File: starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw (hunk 1 above)
Comment (wetted) [reply starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw:827]: @alvarosanchez Do I need to have two `annotationProcessorPaths` each with the different attributes, or am I changing the attribute on the existing one with `combine.self="override"` for all the `<path> child elements?

Comment (wetted) [reply starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/pom.rocker.raw:827]: I think I answered my own question and updated the PR with a revised changeset.

**Review by alvarosanchez — APPROVED** [review]

**Pull request merged**